recipe/meta.yaml

```
@@ -1,55 +1,65 @@
 {% set name = "pipenv" %}
-{% set version = "2022.7.4" %}
-{% set sha256 = "18420fbae2851d2b9d70d1e00f77a8c55d09704a177fe872a0a2da56e98eb018" %}
+{% set version = "2025.0.4" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: {{ sha256 }}
+  url: https://github.com/pypa/{{ name }}/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: d7a481a3a97c954be21e3b4be48a0105fe24c8f02d90875ee026c80af122f9a0
+  patches:
+    - 0001-unix-tests.patch
 
 build:
-  # noarch: python
   number: 0
   entry_points:
     - pipenv=pipenv:cli
     - pipenv-resolver=pipenv.resolver:main
-  script: {{ PYTHON }} -m pip install . -vv
-  skip: true  # [py<37]
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
+  skip: true  # [py<39]
 
 requirements:
   host:
     - pip
     - python
-    - wheel
-    - setuptools
+    - setuptools >=67
   run:
-    - certifi
-    - pip >=22.0.4
     - python
-    - setuptools >=36.2.1
-    - virtualenv
-    - virtualenv-clone >=0.2.5
+    - certifi
+    - packaging >=22
+    # setuptools is a part of run dependency https://github.com/pypa/pipenv/blob/v2025.0.4/pyproject.toml#L31
+    - setuptools >=67
+    - virtualenv >=20.24.2
+  run_constrained:
+    - safety >=3.0.0
 
 test:
+  source_files:
+    - tests
+    - pipenv
+    - run-tests.sh
+    - Pipfile
+    - Pipfile.lock
+    - pyproject.toml
+  imports:
+    - pipenv
+    - pipenv.cli
+    - pipenv.patched
+    - pipenv.routines
+    - pipenv.utils
+    - pipenv.vendor
+  requires:
+    - pip
+    - git
+    - bash # [unix]
```

```suggestion
    - pytest >=5
    - pytest-cov
    - pytest-timeout
    - pytest-xdist
    - mock
    - flaky
    - pytz
```
Adding required test dependencies as per pyproject.toml